platform/mellanox/mlnx-platform-api/sonic_platform/chassis.py

```diff
@@ -1030,3 +1030,165 @@ def get_sfp(self, index):
             return None
 
         return module.get_sfp(sfp_index - 1)
+
+
+class SmartSwitchChassis(Chassis):
+    def __init__(self):
+        super(SmartSwitchChassis, self).__init__()
+        self.module_initialized_count = 0
+        self.module_name_index_map = {}
+        self.initialize_modules()
+
+    def is_modular_chassis(self):
+        """
+        Retrieves whether the sonic instance is part of modular chassis
+
+        Returns:
+            A bool value, should return False by default or for fixed-platforms.
+            Should return True for supervisor-cards, line-cards etc running as part
+            of modular-chassis.
+            For SmartSwitch platforms this should return True even if they are
+            fixed-platforms, as they are treated like a modular chassis as the
+            DPU cards are treated like line-cards of a modular-chassis.
+        """
+        return True
+
+    ##############################################
+    # Module methods
+    ##############################################
+    def initialize_single_module(self, index):
+        count = self.get_num_modules()
+        if index < count:
+            if not self._module_list:
+                self._module_list = [None] * count
+
+            if not self._module_list[index]:
+                from .module import DpuModule
+                self._module_list[index] = DpuModule(index + 1)
+                self.module_name_index_map[self._module_list[index].get_name()] = index
+                self.module_initialized_count += 1
+
+    def initialize_modules(self):
+        count = self.get_num_modules()
+        for index in range(count):
+            self.initialize_single_module(index=index)
+
+    def get_num_modules(self):
+        """
+        Retrieves the number of modules available on this chassis
+        On a SmarSwitch chassis this includes the number of DPUs.
+
+        Returns:
+            An integer, the number of modules available on this chassis
+        """
+        return DeviceDataManager.get_dpu_count()
+
+    def get_all_modules(self):
+        """
+        Retrieves all modules available on this chassis. On a SmarSwitch
+        chassis this includes the number of DPUs.
+
+        Returns:
+            A list of objects derived from ModuleBase representing all
+            modules available on this chassis
+        """
+        self.initialize_modules()
+        return self._module_list
+
+    def get_module(self, index):
+        """
+        Retrieves module represented by (0-based) index <index>
+        On a SmartSwitch index:0 will fetch switch, index:1 will fetch
+        DPU0 and so on
+
+        Args:
+            index: An integer, the index (0-based) of the module to
+            retrieve
+
+        Returns:
+            An object dervied from ModuleBase representing the specified
+            module
+        """
+        self.initialize_single_module(index)
```

Comment:
Description says  "On a SmartSwitch index:0 will fetch switch, index:1 will fetch DPU0 and so on".  Shouldn't the index 0 return a Switch module

Reply:
We are not planning to implement a `switch` module and since the indexing is 0-based, we will return the first dpu for index:0

Reply:
But any upstream logic would expect switch to be under index-0 and that would a problem right. If we are not implementing a switch module then we should return None or a Module() class for index-0.